wire-schema/src/main/java/com/squareup/wire/schema/Extend.java

```diff
@@ -17,26 +17,44 @@
 
 import com.google.common.collect.ImmutableList;
 import com.squareup.wire.schema.internal.parser.ExtendElement;
-import com.squareup.wire.schema.internal.parser.FieldElement;
 
 final class Extend {
-  private final ExtendElement element;
+  private final Location location;
+  private final String documentation;
+  private final String name;
   private final ImmutableList<Field> fields;
   private ProtoType protoType;
 
-  Extend(String packageName, ExtendElement element) {
-    this.element = element;
+  private Extend(String packageName, ExtendElement element) {
+    this.location = element.location();
+    this.documentation = element.documentation();
+    this.name = element.name();
+    this.fields = Field.fromElements(packageName, element.fields(), true);
```

Comment:
Wanna extract this, so you pass in the `List<Field>` ? Otherwise it’s a little asymmetric?